client/src/components/Header/HeaderSearch.tsx

```diff
@@ -1,8 +1,51 @@
-import React, { useState } from 'react';
+import React, { useRef, useState } from 'react';
 import styled from '@emotion/styled';
 
 import { Search } from '@/assets/svgIcons';
 import RecentSearches from './RecentSearches';
+import { setLocalStorageItem } from '@/utils';
+import { RECENT_SEARCH_KEY } from './utils';
+
+const HeaderSearch = (): JSX.Element => {
+  const [isFocused, setFocused] = useState(false);
+  const searchInputRef = useRef<HTMLInputElement>(null);
+
+  const setFocus = () => setFocused(true);
+  const removeFocus = () => setFocused(false);
+
+  const onClick = () => {
+    const name = (searchInputRef.current as HTMLInputElement).value;
+    const date = new Date().toISOString().substr(0, 10).replace(/-/g, '.');
+    const newValue = { date, name };
+
+    setLocalStorageItem(RECENT_SEARCH_KEY, JSON.stringify(newValue));
+    (searchInputRef.current as HTMLInputElement).value = '';
```

Comment:
```suggestion
    if (!searchInputRef.current) return;
    
    const name = searchInputRef.current.value;
    const date = new Date().toISOString().substr(0, 10).replace(/-/g, '.');
    const newValue = { date, name };

    setLocalStorageItem(RECENT_SEARCH_KEY, JSON.stringify(newValue));
    searchInputRef.current.value = '';
```

매번 as를 하는 것 대신에,
이런 식으로도 작성해볼 수 있을 것 같아요.

Reply:
null 체크 코드 추가했습니다 :)